cmake/add_project_python_deps.cmake

```diff
@@ -12,11 +12,32 @@
 set( VIAME_PYTHON_DEPS numpy matplotlib )
 set( VIAME_PYTHON_DEP_CMDS "numpy" "matplotlib" )
 
-if( VIAME_ENABLE_CAMTRAWL )
+if( VIAME_ENABLE_CAMTRAWL OR VIAME_ENABLE_PYTORCH-NETHARN )
   list( APPEND VIAME_PYTHON_DEPS ubelt )
   list( APPEND VIAME_PYTHON_DEP_CMDS "ubelt" )
 endif()
 
+if ( VIAME_ENABLE_PYTORCH-NETHARN )
+  list( APPEND VIAME_PYTHON_DEPS scriptconfig )
+  list( APPEND VIAME_PYTHON_DEP_CMDS "scriptconfig" )
+
+  list( APPEND VIAME_PYTHON_DEPS kwarray )
+  list( APPEND VIAME_PYTHON_DEP_CMDS "kwarray" )
+
+  # Currently only works on linux for now
+  list( APPEND VIAME_PYTHON_DEPS kwimage )
+  list( APPEND VIAME_PYTHON_DEP_CMDS "kwimage" )
+
+  list( APPEND VIAME_PYTHON_DEPS kwplot )
+  list( APPEND VIAME_PYTHON_DEP_CMDS "kwplot" )
+
+  list( APPEND VIAME_PYTHON_DEPS ndsampler )
+  list( APPEND VIAME_PYTHON_DEP_CMDS "ndsampler" )
+
+  list( APPEND VIAME_PYTHON_DEPS netharn )
+  list( APPEND VIAME_PYTHON_DEP_CMDS "netharn" )
+endif()
```

Comment:
Are specifying all of these necessary, or is it only necessary to include netharn and then all of the others are requirements in  netharn's  setup.py/requirements?

Reply:
I could remove kwarray and kwimage, but the other packages are not direct requirements of netharn (they are optional). However, they are needed for this PR. I think I'd prefer to leave kwimage in just due to the issues it will cause on non-linux systems in the near-term future. So I could remove kwarray, but maybe given that the others are necessary it would be better not to and make requirements more explicit?

In general you shouldn't trust that pip will install requirements correctly. It has some mechanism to find a reasonable ordering, but its heuristic based. It is not a general solution. Conda does have a general solution, but that involves solving an integer program, which can be a bit slow.